chips/stm32f429zi/src/ethernet/ethernet.rs

```diff
@@ -0,0 +1,2288 @@
+            0 => false,
+            _ => true,
+        }
+    }
+
+    fn is_mac_address2_enabled(&self) -> bool {
+        match self.mac_registers.maca2hr.read(MACA2HR::AE) {
+            0 => false,
+            _ => true,
+        }
+    }
+
+    fn is_mac_address3_enabled(&self) -> bool {
+        match self.mac_registers.maca3hr.read(MACA3HR::AE) {
+            0 => false,
+            _ => true,
+        }
+    }
+
+    /* === DMA methods === */
+    fn reset_dma(&self) -> Result<(), ErrorCode> {
+        self.dma_registers.dmabmr.modify(DMABMR::SR::SET);
+
+        for _ in 0..100 {
+            if !self.dma_registers.dmabmr.is_set(DMABMR::SR) {
+                return Ok(());
+            }
+        }
+
+        Err(ErrorCode::FAIL)
+    }
+
+    fn dma_transmit_poll_demand(&self) {
+        self.dma_registers.dmatpdr.set(1);
+    }
+
+    fn dma_receive_poll_demand(&self) {
+        self.dma_registers.dmarpdr.set(1);
+    }
+
+    fn set_transmit_descriptor_list_address(&self, address: u32) -> Result<(), ErrorCode> {
+        if self.is_dma_transmission_enabled() == true {
+            return Err(ErrorCode::FAIL);
+        }
+
+        self.dma_registers.dmatdlar.set(address);
+
+        Ok(())
+    }
+
+    fn set_receive_descriptor_list_address(&self, address: u32) -> Result<(), ErrorCode> {
+        if self.is_dma_reception_enabled() == true {
+            return Err(ErrorCode::FAIL);
+        }
+
+        self.dma_registers.dmardlar.set(address);
+
+        Ok(())
+    }
+
+    fn get_transmit_descriptor_list_address(&self) -> u32 {
+        self.dma_registers.dmatdlar.get()
+    }
+
+    fn get_receive_descriptor_list_address(&self) -> u32 {
+        self.dma_registers.dmardlar.get()
+    }
+
+    fn get_transmit_process_state(&self) -> DmaTransmitProcessState {
+        match self.dma_registers.dmasr.read(DMASR::TPS) {
+            0b000 => DmaTransmitProcessState::Stopped,
+            0b001 => DmaTransmitProcessState::FetchingTransmitDescriptor,
+            0b010 => DmaTransmitProcessState::WaitingForStatus,
+            0b011 => DmaTransmitProcessState::ReadingData,
+            0b110 => DmaTransmitProcessState::Suspended,
+            _ => DmaTransmitProcessState::ClosingTransmitDescriptor,
+        }
+    }
+
+    fn get_receive_process_state(&self) -> DmaReceiveProcessState {
+        match self.dma_registers.dmasr.read(DMASR::RPS) {
+            0b000 => DmaReceiveProcessState::Stopped,
+            0b001 => DmaReceiveProcessState::FetchingReceiveDescriptor,
+            0b011 => DmaReceiveProcessState::WaitingForReceivePacket,
+            0b100 => DmaReceiveProcessState::Suspended,
+            0b101 => DmaReceiveProcessState::ClosingReceiveDescriptor,
+            _ => DmaReceiveProcessState::TransferringReceivePacketDataToHostMemory,
+        }
+    }
+
+    fn did_normal_interrupt_occur(&self) -> bool {
+        self.dma_registers.dmasr.is_set(DMASR::NIS)
+    }
```

Comment:
In general, Tock drivers typically use the tock-registers interface directly to query such bits -- accessor methods like these are mostly used in case of some more complex bit-queries / bit-manipulations. However, in the form of private methods they're also totally fine to have around; feel free to leave them in if that's preferred.

Reply:
I find it's easier to understand what code does when using methods with explicit names compared to registers named `dmasr`, especially that the Tock guideline suggests keeping the name of the registers the same as in the reference manual. However, I'm not opposed to remove them if you wish so.

Reply:
It seems fine to keep them, just a style choice.